File: cpp/num_1.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        vector<int> a;
        unordered_map<int, int> m;
        for (int i = 0; i < nums.size(); ++i)
        {
            if (m.find(target - nums[i]) == m.end())
            {
                m[nums[i]] = i;
            }
            else
            {
                a.push_back(m[target - nums[i]]);
                a.push_back(i);
                break;
            }
        }
        return a;
    }
};

File: py/num_1.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        a = []
        d = {}
        for i in range(0, len(nums)):
            if d.get(target - nums[i]) == None:
                d[nums[i]] = i
            else:
                a.append(d[target - nums[i]])
                a.append(i)
                break
        return a
